tests/unit/via/views/route_by_content_test.py

```diff
@@ -94,56 +80,31 @@ def test_sets_correct_cache_control(
         ),
     )
     def test_cache_http_response_codes_appropriately(
-        self, status_code, cache, call_route_by_content, requests
+        self, status_code, cache, call_route_by_content, get_url_details
     ):
-        response = Response()
-        response.status_code = status_code
-        response.raw = mock.Mock()
-
-        requests.get.return_value = response
-
-        result = call_route_by_content("text/html")
+        get_url_details.return_value = (sentinel.content_type, status_code)
+        result = call_route_by_content()
 
         assert result.headers == Any.iterable.containing({"Cache-Control": cache})
 
-    @pytest.mark.parametrize("bad_url", ("no-schema", "glub://example.com", "http://"))
-    def test_it_raises_BadURL_for_invalid_urls(self, bad_url, call_route_by_content):
-        with pytest.raises(BadURL):
-            call_route_by_content(target_url=bad_url)
+    @pytest.fixture(autouse=True)
+    def get_url_details(self, patch):
+        get_url_details = patch("via.views.route_by_content.get_url_details")
+        get_url_details.return_value = (sentinel.content_type, 200)
 
-    @pytest.mark.parametrize(
-        "request_exception,expected_exception",
-        (
-            (MissingSchema, BadURL),
-            (ProxyError, UpstreamServiceError),
-            (SSLError, UpstreamServiceError),
-            (UnrewindableBodyError, UnhandledException),
-        ),
-    )
-    def test_it_catches_requests_exceptions(
-        self, requests, request_exception, expected_exception, call_route_by_content
-    ):
-        requests.get.side_effect = request_exception("Oh noe")
+        return get_url_details
 
-        with pytest.raises(expected_exception):
-            call_route_by_content()
+    @pytest.fixture
+    def pdf_response(self, get_url_details):
+        get_url_details.return_value = ("application/pdf", 200)
 
     @pytest.fixture
-    def requests(self, patch):
-        return patch("via.views.route_by_content.requests")
+    def html_response(self, get_url_details):
+        get_url_details.return_value = ("application/html", 200)
 
     @pytest.fixture
     def call_route_by_content(self, make_request):
-        def call_route_by_content(
-            content_type="text/html", target_url="http://example.com", params=None
-        ):
-            httpretty.register_uri(
-                method=httpretty.GET,
-                uri=urlparse(target_url)._replace(query=None, fragment=None).geturl(),
-                body="DUMMY",
-                adding_headers={"Content-Type": content_type},
-            )
```

Comment:
We don't need to mock HTTP calls any more as we now mock `get_url_details` as an external dependency.